### PR TITLE
refactor(Response): rename Response.body to Response.text

### LINKED ISSUE #1578: Rename resp.body to resp.text
For both the WSGI and ASGI interfaces, rename Response.body to Response.text to provide a better contrast to Response.data. Response.body should then be made a deprecated alias.

Don't forget to also grep the docstrings and /docs for references to 'body' and update them accordingly.

NOTE: It may be best to wait to implement this issue until #1573 is merged to avoid conflicts.

### PR DESCRIPTION
**Pull request by ashutoshvarma**
# Summary of Changes

This PR renames the `Response.body` to `Response.text` in both interfaces. Also `Response.body` is made deprecated alias for new `Response.text`
- All the references are changed in both docs and docstrings
- getters and setters for `Response.body` has been decorated with `deprecated(is_property=True)`. I was not sure about deprecation message, so I just wrote what came to my mind first.
- tests for `Response.body` are not removed from `test_response_media_asgi.py` and `test_response_media.py` instead same test has been added for `Response.text`

# Related Issues

Fix #1578 

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] Applied changes to both WSGI and ASGI code paths and interfaces (where applicable).
- [x] Added **tests** for changed code.
- [x] Prefixed code comments with GitHub nick and an appropriate prefix.
- [x] Coding style is consistent with the rest of the framework.
- [x] Updated **documentation** for changed code.
    - [x] Added docstrings for any new classes, functions, or modules.
    - [x] Updated docstrings for any modifications to existing code.
    - [x] Updated both WSGI and ASGI docs (where applicable).
    - [x] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [x] Updated all relevant supporting documentation files under `docs/`.
    - [x] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [x] Changes (and possible deprecations) have [towncrier](https://towncrier.readthedocs.io/en/actual-freaking-docs/index.html) news fragments under `docs/_newsfragments/`, with the file name format `{issue_number}.{fragment_type}.rst`. (Run `towncrier --draft` to ensure it renders correctly.)

### CODE REVIEW

File: docs/user/faq.rst [hunk 1]
```diff
@@ -783,15 +783,15 @@ rest.
         def on_get(self, req, resp):
             resp.media = { 'hello': 'World' }
 
-`resp.body` and `resp.data` are very similar, they both allow you to set the
+`resp.text` and `resp.data` are very similar, they both allow you to set the
 body of the response. The difference being, `body` takes a string and `data`
```
Comment (CaselIT) [inline docs/user/faq.rst:787]: Missed body: "The difference being, `body` takes a string"

File: falcon/app.py [hunk 2]
```diff
@@ -872,8 +872,8 @@ def _compose_status_response(self, req, resp, http_status):
             resp.set_headers(http_status.headers)
 
         # NOTE(kgriffs): If http_status.body is None, that's OK because
-        # it's acceptable to set resp.body to None (to indicate no body).
-        resp.body = http_status.body
+        # it's acceptable to set resp.text to None (to indicate no body).
+        resp.text = http_status.body
```
Comment (CaselIT) [inline falcon/app.py:876]: shouldn't we also use `http_status.text`

**Review by CaselIT — COMMENTED** [review]
Looks good. There may still be some `body` mention lingering about though

While le issue did not mention `HttpStatus`, I think that it should also rename `body` -> `text`

File: falcon/app.py (hunk 2 above)
Comment (kgriffs) [reply falcon/app.py:876]: +1 for this. But please also retain `body` as a kwarg and also provide a `body` property for backward compatibility. The `HTTPStatus` docstring should mention that the kwarg and property are deprecated aliases and will be removed in a future release.

File: falcon/asgi/response.py [hunk 3]
```diff
@@ -50,20 +50,22 @@ class Response(falcon.response.Response):
                 See also :ref:`media` for more information regarding media
                 handling.
 
-        body (str): String representing response content.
```
Comment (kgriffs) [inline falcon/asgi/response.py:53]: It would be helpful to still include this in the docstring, yet note that it is a deprecated alias.

**Review by vytas7 — CHANGES_REQUESTED** [review]
We ought to also keep a couple of tests referencing the deprecated to property in order to make sure it continues to work as expected.

Furthermore, as you have correctly indicated in the PR checklist, Towcrier fragment(s) are still missing for the change.

**Comment by vytas7** [comment]
@ashutoshvarma just checking if you might have a chance to continue working on this (by addressing @kgriffs and @CaselIT comments) within the next couple of weeks?

**Comment by ashutoshvarma** [comment]
Sorry it took me so long to respond, I had been busy with stuff. 
Yeah I will try to work on changes suggested by @CaselIT and @kgriffs by this week.

**Comment by vytas7** [comment]
No worries @ashutoshvarma, and thanks for the update.

**Comment by vytas7** [comment]
Hi @ashutoshvarma !
Sorry for persistence, but just checking again... the same question :smiling_imp:

**Comment by CaselIT** [comment]
Otherwise if you are ok with it, we could take this to the finish line :)

**Comment by ashutoshvarma** [comment]
Terribly sorry for not fulfilling my commitment.
You can go ahead with it, I don't think I will be able to complete this.

**Comment by vytas7** [comment]
Heh, no worries @ashutoshvarma !
As a volunteer devoting free time to help an open source project, you have no commitments to us!

@CaselIT would you like to take this over the finish line?

**Comment by maknop** [comment]
Is someone still working on this? I'm new to contributing so I thought this would be interesting to tackle if it isn't still taken.

**Comment by vytas7** [comment]
> Is someone still working on this? I'm new to contributing so I thought this would be interesting to tackle if it isn't still taken.

(As I understand this was a limited Black Weekend offer that has now expired :smiling_imp: .)

@maknop Yes, you are very welcome to take over if you still wish to!

**Comment by CaselIT** [comment]
> Heh, no worries @ashutoshvarma !
> As a volunteer devoting free time to help an open source project, you have no commitments to us!
> 
> @CaselIT would you like to take this over the finish line?

I may be able to work a bit on it this week

**Comment by CaselIT** [comment]
A bit late, but I've managed to come around and finish this

**Review by CaselIT — DISMISSED** [review]

File: tests/test_alias.py [hunk 4]
```diff
@@ -14,7 +15,8 @@ def on_get(self, req, resp):
 
 @pytest.fixture
 def alias_client():
-    api = falcon.API()
+    with pytest.warns(DeprecatedWarning, match='API class may be removed'):
```
Comment (vytas7) [inline tests/test_alias.py:18]: These are neat, I'm not sure why it wasn't done this way in the first place.

**Review by vytas7 — CHANGES_REQUESTED** [review]
This is looking good :+1: 

I could, however, `grep` a couple of items that could be revised too:
* `docs/_newsfragments/1679.breakingchange.rst`
* `docs/user/recipes/output-csv.rst`

File: tests/test_alias.py (hunk 4 above)
Comment (CaselIT) [reply tests/test_alias.py:18]: not sure, I was chasing a warning, so I fixed a bunch along the way

**Review by CaselIT — APPROVED** [review]

**Review by kgriffs — APPROVED** [review]
Looks good!

**Review by vytas7 — APPROVED** [review]

**Pull request merged**